docs/src/intro.md

```diff
@@ -22,9 +22,60 @@ The goal of this project is to create a standardized, generic, modular framework
 
 The intention is to integrate many existing 2nix converters into this framework, thereby improving many of the previously named aspects and providing a unified UX for all 2nix solutions.
 
+
+### Modularity:
```

Comment:
moved here from `architectural-considerations.md` also fits nicely with the sections above.